js/ui.js

```diff
@@ -0,0 +1,586 @@
+            name.classList.add("legendary")        
+        else 
+            name.classList.remove("legendary")        
+
+        const active = row.getElementsByClassName("active")[0]
+        const color = autoBuyEnabled
+            ? itemCategories["Properties"].includes(item.name) ? headerRowColors["Properties_Auto"] : headerRowColors["Misc_Auto"]
+            : itemCategories["Properties"].includes(item.name) ? headerRowColors["Properties"] : headerRowColors["Misc"]
+
+        active.style.backgroundColor = gameData.currentMisc.includes(item) || item == gameData.currentProperty ? color : "white"
+        row.getElementsByClassName("effect")[0].textContent = item.getEffectDescription()
+        formatCoins(item.getExpense(), row.getElementsByClassName("expense")[0])
+    }
+}
+
+function updateHeaderRows(categories) {
+    for (const categoryName in categories) {
+        const className = removeSpaces(categoryName)
+        const headerRow = document.getElementsByClassName(className)[0]
+        const maxLevelElement = headerRow.getElementsByClassName("maxLevel")[0]
+        gameData.rebirthOneCount > 0 ? maxLevelElement.classList.remove("hidden") : maxLevelElement.classList.add("hidden")
+    }
+}
+
+function updateText() {
+    //Sidebar
+    document.getElementById("ageDisplay").textContent = formatAge(gameData.days)
+    document.getElementById("lifespanDisplay").textContent = format(daysToYears(getLifespan()))
+    document.getElementById("realtimeDisplay").textContent = formatTime(gameData.realtime)
+    document.getElementById("pauseButton").textContent = gameData.paused ? "Play" : "Pause"
+
+    formatCoins(gameData.coins, document.getElementById("coinDisplay"))
+    setSignDisplay()
+    formatCoins(getNetIncome(), document.getElementById("netDisplay"))
+    formatCoins(getIncome(), document.getElementById("incomeDisplay"))
+    formatCoins(getExpense(), document.getElementById("expenseDisplay"))
+
+    document.getElementById("happinessDisplay").textContent = format(getHappiness())
+
+    document.getElementById("evilDisplay").textContent = format(gameData.evil)
+    document.getElementById("evilGainDisplay").textContent = format(getEvilGain())
+    document.getElementById("evilGainButtonDisplay").textContent = "+" + format(getEvilGain())
+
+    document.getElementById("essenceDisplay").textContent = format(gameData.essence)
+    document.getElementById("essenceGainDisplay").textContent = format(getEssenceGain())
+    document.getElementById("essenceGainButtonDisplay").textContent = "+" + format(getEssenceGain())
+
+    // TODO Why don't we use getGameSpeed()
+    document.getElementById("timeWarpingDisplay").textContent = "x" + format(
+        gameData.taskData["Time Warping"].getEffect() *
+            gameData.taskData["Temporal Dimension"].getEffect() *
+            gameData.taskData["Time Loop"].getEffect() *
+            (gameData.requirements["Eternal Time"].isCompleted() ? 2 : 1)
+        )
```

Comment:
`getGameSpeed()` returns 0 if age &ge; lifespan or if the game is paused.

The function could be split into two: `getUnpausedGameSpeed()` which is the same as current `getGameSpeed()` but without the `+!gameData.paused` and `+isAlive()` parts, and a new `getGameSpeed()` which just returns `getUnpausedGameSpeed()` if `!gameData.paused && isAlive()`.

The UI code above can just use `getUnpausedGameSpeed()` in place of what is effectively a copy of its content.

Reply:
Agreed. Will implement in a future refactor!